hyperlane/cmd/hyp/cmd/hyperlane.go

```diff
@@ -0,0 +1,200 @@
+	}
+
+	fmt.Printf("successfully got block %d from ev-reth\n", block.NumberU64())
+
+	namespace, err := hex.DecodeString(namespaceHex)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	pubKey, err := getSequencerPubKey(ctx, evnodeClient)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("successfully got pubkey from ev-node %x\n", pubKey)
+
+	groth16Vkey := readGroth16Vkey()
+	stateTransitionVkey := readStateTransitionVkey()
+	stateMembershipVkey := readStateMembershipVkey()
+
+	msgCreateZkExecutionISM := zkismtypes.MsgCreateZKExecutionISM{
+		Creator:             broadcaster.address.String(),
+		StateRoot:           block.Header().Root.Bytes(),
+		Height:              block.NumberU64(),
+		Namespace:           namespace,
+		SequencerPublicKey:  pubKey,
+		Groth16Vkey:         groth16Vkey,
+		StateTransitionVkey: stateTransitionVkey,
+		StateMembershipVkey: stateMembershipVkey,
+	}
+
+	res := broadcaster.BroadcastTx(ctx, &msgCreateZkExecutionISM)
+	ismID := parseIsmIDFromZkISMEvents(res.Events)
+
+	return ismID
+}
+
+// SetupWithIsm deploys the cosmosnative Hyperlane components using the provided ism identifier.
+func SetupWithIsm(ctx context.Context, broadcaster *Broadcaster, ismID util.HexAddress) {
+	msgCreateNoopHooks := hooktypes.MsgCreateNoopHook{
+		Owner: broadcaster.address.String(),
+	}
+
+	res := broadcaster.BroadcastTx(ctx, &msgCreateNoopHooks)
+	hooksID := parseHooksIDFromEvents(res.Events)
+
+	msgCreateMailBox := coretypes.MsgCreateMailbox{
+		Owner:        broadcaster.address.String(),
+		DefaultIsm:   ismID,
+		LocalDomain:  69420,
+		DefaultHook:  &hooksID,
+		RequiredHook: &hooksID,
+	}
+
+	res = broadcaster.BroadcastTx(ctx, &msgCreateMailBox)
+	mailboxID := parseMailboxIDFromEvents(res.Events)
+
+	msgCreateCollateralToken := warptypes.MsgCreateCollateralToken{
+		Owner:         broadcaster.address.String(),
+		OriginMailbox: mailboxID,
+		OriginDenom:   denom,
+	}
+
+	res = broadcaster.BroadcastTx(ctx, &msgCreateCollateralToken)
+	tokenID := parseCollateralTokenIDFromEvents(res.Events)
+
+	// set ism id on new collateral token (for some reason this can't be done on creation)
+	msgSetToken := warptypes.MsgSetToken{
+		Owner:    broadcaster.address.String(),
+		TokenId:  tokenID,
+		IsmId:    &ismID,
+		NewOwner: broadcaster.address.String(),
+	}
+
+	broadcaster.BroadcastTx(ctx, &msgSetToken)
+
+	cfg := &HyperlaneConfig{
+		IsmID:     ismID,
+		HooksID:   hooksID,
+		MailboxID: mailboxID,
+		TokenID:   tokenID,
+	}
+
+	writeConfig(cfg)
+}
+
+// SetupRemoteRouter links the provided token identifier on the cosmosnative deployment with the receiver contract on the counterparty.
+// For example: if the provided token identifier is a collateral token (e.g. utia), the receiverContract is expected to be the
+// contract address for the corresponding synthetic token on the counterparty.
+func SetupRemoteRouter(ctx context.Context, broadcaster *Broadcaster, tokenID util.HexAddress, domain uint32, receiverContract string) {
```

Comment:
why don't we leverage the hyperlane CLI instead?

Reply:
does hyperlane CLI support deploying on cosmos now? 

I created this custom CLI for the purposes of this repo because there was no support and no docs for cosmosnative deployments when I started working on this - so I had to figure out how to wire everything up manually.

I think it is best to keep this right now for our development, because even if the official CLI supports deployment on cosmosnative it is still undocumented and does not allow us to deploy our custom zk ism.